### PR TITLE
feat(wallet-ffi): adds FFI wallet_get_utxos() and test_wallet_get_utxos()

### PR DESCRIPTION
**Pull request by agubarev**
Motivation and Context
---
To allow client app (mobile) to get a paginated list of unspent outputs, filtered by a minimum value threshold.

How Has This Been Tested?
---
unit test

### CODE REVIEW

File: base_layer/wallet_ffi/src/lib.rs [hunk 1]
```diff
@@ -4093,6 +4106,99 @@ pub unsafe extern "C" fn wallet_get_balance(wallet: *mut TariWallet, error_out:
...
+    debug!(
+        target: LOG_TARGET,
+        "page = {:#?} page_size = {:#?} sort_asc = {:#?} dust_threshold = {:#?}",
+        page,
+        page_size,
+        sort_ascending,
+        dust_threshold
+    );
+
+    let rt = match Runtime::new() {
```
Comment (sdbondi) [inline base_layer/wallet_ffi/src/lib.rs]: The wallet already has a runtime (it's an anti-pattern to create multiple runtimes)

`wallet.runtime`

**Review by sdbondi — COMMENTED** [review]
Nice looks good - just one change RE the runtime.
Please also add details to the PR description

File: base_layer/wallet_ffi/src/lib.rs [hunk 2]
```diff
@@ -4093,6 +4106,99 @@ pub unsafe extern "C" fn wallet_get_balance(wallet: *mut TariWallet, error_out:
...
+        Err(e) => {
+            error = LibWalletError::from(InterfaceError::TokioError(e.to_string())).code;
+            ptr::swap(error_out, &mut error as *mut c_int);
+            return ptr::null_mut();
+        },
+    };
+
+    match rt.block_on((*wallet).wallet.output_manager_service.get_unspent_outputs()) {
+        Ok(mut unblinded_outputs) => {
+            unblinded_outputs.sort_by(|a, b| match sort_ascending {
```
Comment (sdbondi) [inline base_layer/wallet_ffi/src/lib.rs]: Ideally ordering and pagination should happen in the DB layer but we can refactor that later

File: base_layer/wallet_ffi/src/lib.rs (hunk 1 above)
Comment (agubarev) [reply base_layer/wallet_ffi/src/lib.rs]: Ah, that's right, removed new runtime initialization and using:
```rust
    match (*wallet)
        .runtime
        .block_on((*wallet).wallet.output_manager_service.get_unspent_outputs())
    {
        Ok(mut unblinded_outputs) => {
```

like in other places, thanks!

File: base_layer/wallet_ffi/src/lib.rs (hunk 2 above)
Comment (agubarev) [reply base_layer/wallet_ffi/src/lib.rs]: Yes, I have it planned after I get a better grasp of the project basics.

**Review by stringhandler — APPROVED** [review]

**Pull request merged**